### PR TITLE
OPCUA-1020 client side methods

### PR DESCRIPTION
**Pull request by pnikiel**
Task driven by FPGA configuration for NSW et al.

It introduces client-side method calling (no relation to open62541-client method support for servers! that is a separate mechanism).

Tested with calling of big ByteString (52MB of size), worked fine, no single warning from valgrind.
Wrong status codes seem well handled.

Aspects not tested: return values passing.

### CODE REVIEW

**Comment by ben-farnham** [comment]
Bit lost as to what this feature is - thus, struggling to review

> no relation to open62541-client method support for servers! that is a separate mechanism

- What is _open62541-client_ ? Typo for open62541-compat?

**Comment by pnikiel** [comment]
Sorry, it was a typo:
this feature is support for calling methods for OPC-UA **clients** based on open62541-compat.

File: src/uaclient/uasession.cpp [hunk 1]
```diff
@@ -231,16 +231,95 @@ UaStatus UaSession::write(
...
 
     return writeResponse.responseHeader.serviceResult;
 }
 
 
+/* What's not implemented:
+ * - diagnosticInfos
+ */
+UaStatus UaSession::call(
+        ServiceSettings &       serviceSettings,
```
Comment (ben-farnham) [inline src/uaclient/uasession.cpp:255]: Am I missing something? Why do we keep passing serviceSettings around but not apparently using it?

Comment (pnikiel) [reply src/uaclient/uasession.cpp:255]: That is correct. open62541, as for as I understand, doesn't yet support per-call ServiceSettings and it uses the global one (which has been made configurable in the previous pull request).
On the other hand we have to keep the API this way to cover both open62541 and UASDK.
I think open62541 will some day support per-call ServiceSettings.

Comment (ben-farnham) [reply src/uaclient/uasession.cpp:255]: Ah, an API equivalence thing then. OK

Comment (pnikiel) [reply src/uaclient/uasession.cpp:255]: Would you have some further comments/requests?

Comment (ben-farnham) [reply src/uaclient/uasession.cpp:255]: Nope - don't think I do. All ready to merge I think.

Comment (pnikiel) [reply src/uaclient/uasession.cpp:255]: thanks!

**Pull request merged**